Raspbian solution

GL513 opened the pull request:
Feel free to criticize.

GL513 committed:
Update README.md

GL513 committed:
Update README.md

GL513 committed:
Update README.md

GL513 committed:
Update README.md

ebarrabbb commented:
Hi! I have compiled and run it on Raspberry Pi 2 and it run but it does no mine. I think it is too slow. There wasn't any "Accept" job from Flockpool. I tested it for more than an hour. You could inform about it or test it to check the issue. Thanks anyway for your version.

ebarrabbb commented:
Raspberry Pi 2: run but don't mine

I have compiled and run it on Raspberry Pi 2 and it run but it does not mine. I think it is too slow. There wasn't any "Accept" job from Flockpool. I tested it for more than an hour. You could inform about it or test it to check the issue. Thanks anyway for your version.

GL513 committed:
Update cpu-miner.c

GL513 commented:
dont merge the cpu-miner.c file, i modified it to test something (i promise i am not and never will distribute)

shouldve made a new branch for that...

GL513 commented:
> Raspberry Pi 2: run but don't mine
> 
> I have compiled and run it on Raspberry Pi 2 and it run but it does not mine. I think it is too slow. There wasn't any "Accept" job from Flockpool. I tested it for more than an hour. You could inform about it or test it to check the issue. Thanks anyway for your version.

sometimes flockpool is weird, ive noticed. try giving it an hour or two and it should appear

ldamato commented:
Hi,
I'm experiencing this issue.

root@raspberrypi:/usr/src/cpuminer-gr# ./build.sh
make: *** No rule to make target 'distclean'.  Stop.
./autogen.sh: 8: aclocal: not found
Detected ARMv7 (arm) system
Detected vfpv4 instruction set. Changing to -mfpu=neon-vfpv4
configure: error: cannot find install-sh, install.sh, or shtool in "." "./.." "./../.."
make: *** No targets specified and no makefile found.  Stop.
strip: 'cpuminer': No such file

GL513 committed:
Update cpu-miner.c

removed fee change and replaced with original code, branched for fee change on fork

ldamato commented:
Hi, I guess I found the same issue on Raspberry Pi3 B+

[2021-11-17 15:17:54] gr: eu.flockpool.com:5555
                      Periodic Report     5m00s        21m28s
                      Share rate        0.00/min     0.00/min
                      Hash rate          0.00h/s      0.00h/s   (8.17h/s)
                      Submitted             0            0
                      Accepted              0            0        0.0%
                      Hi/Lo Share Diff  0 /  9e+99
[2021-11-17 15:17:54] New Work: Block 186455, Net diff 7.5162, Job 00000193-0
[2021-11-17 15:18:07] CPU temp: curr 55 C max 55, Freq: 1.200/1.200 GHz
[2021-11-17 15:18:07] New Block 186456, Job 00000194-0
                      Diff: Net 7.2523, Stratum 0.5, Target 7.6294e-06
                      TTF @ 8.11 h/s: Block 68y68d, Share 1h07m
                      Stratum Diff 0.5, Block 186456, Job 00000194-0

I'll keep you updated if something new appears

cybercuate commented:
I use Oracle OCI and their Ampere instances use ARM64 and after a while runnig fine I get **_"stratum_recv_line failed"_** and stops mining. So I have to login to the instance and just run it again. Is there a solution for this? like running automatically again every time it stops?

GL513 commented:
> I use Oracle OCI and their Ampere instances use ARM64 and after a while runnig fine I get **_"stratum_recv_line failed"_** and stops mining. So I have to login to the instance and just run it again. Is there a solution for this? like running automatically again every time it stops?

on raspbain, with this shell, i did notice with low pings it will "crash".
try doing something like `./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json`
and so on, using `;` as the indicator that tells linux, "do this next", in short.

ldamato commented:
I tried ./cpuminer -c config.json ; ./cpumier -c config.json ; etc...
it looks that I don't have config.json
unable to open config.json: No such file or directory
./cpuminer: no algo supplied
Try `--help' for more information.
bash: ./cpumier: No such file or directory

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

unable to open config.json: No such file or directory
./cpuminer: no algo supplied
Try `--help' for more information.
bash: ./cpumier: No such file or directory

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

unable to open config.json: No such file or directory



EDIT:
I tried 

./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ;

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

[2021-11-17 20:04:28] Unknown algo: x11
Try `--help' for more information.

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

[2021-11-17 20:04:29] Unknown algo: x11
Try `--help' for more information.

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

[2021-11-17 20:04:29] Unknown algo: x11
Try `--help' for more information.

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

[2021-11-17 20:04:29] Unknown algo: x11
Try `--help' for more information.

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

[2021-11-17 20:04:29] Unknown algo: x11
Try `--help' for more information.

         **********  cpuminer-opt-gr 3.9.6.1  ***********
     A CPU miner with multi algo support and optimized for CPUs
     with AVX512, SHA and VAES extensions by JayDDee.
     BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT

[2021-11-17 20:04:29] Unknown algo: x11
Try `--help' for more information.

GL513 commented:
> I tried ./cpuminer -c config.json ; ./cpumier -c config.json ; etc... it looks that I don't have config.json unable to open config.json: No such file or directory ./cpuminer: no algo supplied Try `--help' for more information. bash: ./cpumier: No such file or directory
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> unable to open config.json: No such file or directory ./cpuminer: no algo supplied Try `--help' for more information. bash: ./cpumier: No such file or directory
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> unable to open config.json: No such file or directory
> 
> EDIT: I tried
> 
> ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ; ./cpuminer -c cpuminer-conf.json ;
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> [2021-11-17 20:04:28] Unknown algo: x11 Try `--help' for more information.
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> [2021-11-17 20:04:29] Unknown algo: x11 Try `--help' for more information.
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> [2021-11-17 20:04:29] Unknown algo: x11 Try `--help' for more information.
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> [2021-11-17 20:04:29] Unknown algo: x11 Try `--help' for more information.
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> [2021-11-17 20:04:29] Unknown algo: x11 Try `--help' for more information.
> 
> ```
>      **********  cpuminer-opt-gr 3.9.6.1  ***********
>  A CPU miner with multi algo support and optimized for CPUs
>  with AVX512, SHA and VAES extensions by JayDDee.
>  BTC donation address: 12tdvfF7KmAsihBXQXynT6E6th2c2pByTT
> ```
> 
> [2021-11-17 20:04:29] Unknown algo: x11 Try `--help' for more information.

are you new to raptoreum, or thhe cpuminer?
you need to put a config file, like you can make [here](https://flockpool.com/start-mining) and put that in the cpuminer-gr folder, then try it again

ldamato commented:
:) Definitely n00b. Started yesterday. Now I got it! It works: thanks!

I added the string  ./cpuminer -a gr -o stratum+tcps://eu.flockpool.com:5555 -u etc, etc... in the rc.local and I configured a restart every hour in crontab.

I guess that if I quit ssh... mmmmh... the process will be killed

GL513 commented:
> :) Definitely n00b. Started yesterday. Now I got it! It works: thanks!
> 
> I added the string  ./cpuminer -a gr -o stratum+tcps://eu.flockpool.com:5555 -u etc, etc... in the rc.local and I configured a restart every hour in crontab.
> 
> I guess that if I quit ssh... mmmmh... the process will be killed

*Stupid analogy time.*

No, if ssh is quit, the commands will still work because it's like putting peanut butter on bread. Once it's there, it's there. You don't have to wait for the peanut butter to be on the bread to eat it though, you could just jam you hand in there and eat it off your hand.

GL513 committed:
Update README.md

digitalincome commented:
> > I use Oracle OCI and their Ampere instances use ARM64 and after a while runnig fine I get **_"stratum_recv_line failed"_** and stops mining. So I have to login to the instance and just run it again. Is there a solution for this? like running automatically again every time it stops?
> 
> on raspbain, with this shell, i did notice with low pings it will "crash". try doing something like `./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json ; ./cpuminer -c config.json ; ./cpumier -c config.json` and so on, using `;` as the indicator that tells linux, "do this next", in short.

Hello,
I also have the same issue running the miner on Oracle - "stratum_recv_line failed".
But I noticed that it appears every time after developer fee is applied (see screenshot). Checked on two different pools - flockpool and rplant, and reproduces on both of them.
P.s. The first fee applies well, but the second one always fails.
Appreciate any advice.
![Screenshot 2021-11-29 005922](https://user-images.githubusercontent.com/72601785/143945621-3e2440bb-d6c8-461b-bf26-a973073964a0.png)

**_Update_**:
Found a workaround using bash:

> for i in {0..999999}
> do
> ./cpuminer -c cpuminer-conf.json
> done

tyas0909 commented:
add to ARM
